app/store/sync.ts

```diff
@@ -15,99 +15,193 @@ import { createSyncClient, ProviderType } from "../utils/cloud";
 import { corsPath } from "../utils/cors";
 
 export interface WebDavConfig {
-  server: string;
+  endpoint: string;
   username: string;
   password: string;
+  filename: string;
 }
 
-export type SyncStore = GetStoreState<typeof useSyncStore>;
+export interface GistConfig {
+  filename: string;
+  gistId: string;
+  token: string;
+}
 
-export const useSyncStore = createPersistStore(
-  {
-    provider: ProviderType.WebDAV,
-    useProxy: true,
-    proxyUrl: corsPath(ApiPath.Cors),
-
-    webdav: {
-      endpoint: "",
-      username: "",
-      password: "",
-    },
+export type SyncStore = GetStoreState<typeof useSyncStore> & {
+  syncing: boolean;
+};
 
-    upstash: {
-      endpoint: "",
-      username: "",
-      apiKey: "",
-    },
+const DEFAULT_SYNC_STATE = {
+  provider: ProviderType.WebDAV,
+  useProxy: true,
+  proxyUrl: corsPath(ApiPath.Cors),
+  enableAccessControl: false,
+
+  githubGist: {
+    filename: "",
+    gistId: "",
+    token: "",
+  },
+
+  webdav: {
+    endpoint: "",
+    username: "",
+    password: "",
+    filename: "",
+  },
 
-    lastSyncTime: 0,
-    lastProvider: "",
+  upstash: {
+    endpoint: "",
+    username: "",
+    apiKey: "",
+    filename: "",
   },
+
+  lastSyncTime: 0,
+  lastProvider: "",
+  lastUpdateTime: 0,
+  syncing: false,
+};
+export const useSyncStore = createPersistStore(
+  DEFAULT_SYNC_STATE,
   (set, get) => ({
-    coundSync() {
+    countSync() {
       const config = get()[get().provider];
       return Object.values(config).every((c) => c.toString().length > 0);
     },
 
-    markSyncTime() {
-      set({ lastSyncTime: Date.now(), lastProvider: get().provider });
+    markSyncTime(provider: ProviderType) {
+      set({ lastSyncTime: Date.now(), lastProvider: provider });
+    },
+
+    markUpdateTime() {
+      set({ lastUpdateTime: Date.now() });
     },
 
     export() {
       const state = getLocalAppState();
-      const fileName = `Backup-${new Date().toLocaleString()}.json`;
-      downloadAs(JSON.stringify(state), fileName);
+      const fileName = `Backup-${new Date().toLocaleString()}`;
```

Comment:
Is this change necessary? Can we just keep `downloadAs` receive string?

Reply:
> Is this change necessary? Can we just keep `downloadAs` receive string?

that branch are outdated
a better one already implemented [here](https://github.com/H0llyW00dzZ/ChatGPT-Next-Web/blob/main/app/store/sync.ts#L92):
```ts
    export() {
      const state = getLocalAppState();
      const datePart = isApp
        ? `${new Date().toLocaleDateString().replace(/\//g, '_')} ${new Date().toLocaleTimeString().replace(/:/g, '_')}`
        : new Date().toLocaleString();

      const fileName = `Backup-${datePart}.json`;
      downloadAs((state), fileName);
    },
 ```